pipelines/rbsa/rbsa.py

```diff
@@ -80,68 +80,117 @@ def generate_result_plots(self):
+            city_df = enduse_loadshapes.loc[enduse_loadshapes.target == city]
+            max_total = city_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
+            if max_total <= 1: max_val = 1
+            else: max_val = int(max_total) + 1
+            for ydx, daytype in enumerate(city_df.daytype.unique()):
+                title = f'{str(city)}-{str(daytype)}'
+                day_df = city_df.loc[city_df.daytype == daytype]
+                day_df = day_df.append(day_df.iloc[0])
+                day_df = day_df.reset_index()
+                day_df['Baseload'] = day_df[base_enduses].sum(axis=1)
+                plot = day_df[['Baseload', 'Heating', 'Cooling']].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['black','red','blue'])
+                fig = plot.get_figure()
+                image_index_based_name = '{0:0=2d}'.format(image_index)
+                fig.savefig(f'{enduse_plots_dir}/{image_index_based_name}.png')
+                plt.close(fig)
+                image_index += 1
 
         logger.info('GENERATING TOTAL LOADSHAPE PLOTS')
 
         image_index = 0
         for idx, city in enumerate(total_loadshapes.target.unique()):
             city_df = total_loadshapes.loc[total_loadshapes.target == city]
-            
+            max_total = city_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
+            if max_total <= 1: max_val = 1
+            else: max_val = int(max_total) + 1
             for ydx, daytype in enumerate(city_df.daytype.unique()):
                 title = f'{str(city)}-{str(daytype)}'
                 day_df = city_df.loc[city_df.daytype == daytype]
+                day_df = day_df.append(day_df.iloc[0])
                 day_df = day_df.reset_index()
-                day_df['Baseload'] = day_df[total_base_enduses].sum(axis=1)
-                plot = day_df[['Heating','Cooling','Baseload']].plot(title=title, grid=True, xticks=ticks, ylim=(0, 1), linewidth=2, color=['red','blue','black'])
+                day_df['Baseload'] = day_df[base_enduses].sum(axis=1)
+                plot = day_df[['Baseload', 'Heating', 'Cooling']].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['black','red','blue'])
                 fig = plot.get_figure()
                 image_index_based_name = '{0:0=2d}'.format(image_index)
                 fig.savefig(f'{total_plots_dir}/{image_index_based_name}.png')
+                plt.close(fig)
                 image_index += 1
+
+        logger.info('GENERATING LOADSHAPE PLOTS')
+
+        image_index = 0
+        for idx, city in enumerate(loadshapes.zipcode.unique()):
+            city_df = loadshapes.loc[loadshapes.zipcode == city]
+            max_total = city_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
+            if max_total <= 1: max_val = 1
+            else: max_val = int(max_total) + 1
```

Comment:
There's a lot of areas where this code can be simplified. I put the suggestion somewhere else, however, other than that, this PR looks good.

Reply:
Fixed all of it and added ceus plot labels